File: mixin-base/src/main/java/net/yakclient/mixin/base/internal/instruction/DirectInsnInterceptor.java
package net.yakclient.mixin.base.internal.instruction;

import net.yakclient.mixin.base.YakMixins;
import org.objectweb.asm.tree.MethodNode;

public class DirectInsnInterceptor extends MethodNode implements InstructionInterceptor {
    public DirectInsnInterceptor() {
        super(YakMixins.ASM_VERSION);
    }

    @Override
    public Instruction intercept() {
        return new DirectInstruction(this.instructions);
    }
}

File: base/src/main/kotlin/net/yakclient/mixins/base/internal/instruction/DirectInsnInterceptor.kt
package net.yakclient.mixins.base.internal.instruction

import net.yakclient.mixins.base.internal.bytecode.ByteCodeUtils
import org.objectweb.asm.tree.MethodNode

open class DirectInsnInterceptor : MethodNode(ByteCodeUtils.ASM_VERSION), InstructionInterceptor {
    override fun intercept(): Instruction {
        return DirectInstruction(instructions)
    }
}